### PR TITLE
README.md file has been updated. "Set up the project" and "Key Featur…

### PR DESCRIPTION
**Pull request by MortezaGitPro**
README.md file has been updated. "Set up the project" and "Key Features" have been added.
Issue reference: Update README.md
#50 opened 4 hours ago by MortezaGitPro
https://github.com/MajorTreble/WhateverJobYouChoose-Soen6011summer2023/issues/50

### CODE REVIEW

**Review by nioushamahjoub — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -27,9 +27,54 @@ The application is a career search web tool to match students looking for jobs w
   - Role: Full-Stack Developer
 
 ## Project Approach and Technology
-The team is using a Scrum based approach with regular sprints. The project has a backlog of user stories that have been broken down into tasks and estimated. With the help of a product owner, the stories are prioritized and put into a sprint. 
+The team is using a Scrum-based approach with regular sprints. The project has a backlog of user stories that have been broken down into tasks and estimated. With the help of a product owner, the stories are prioritized and put into a sprint.
```
Comment (MajorTreble) [inline README.md:30]: There is no dash between scrum and based

File: README.md [hunk 2]
```diff
@@ -27,9 +27,54 @@ The application is a career search web tool to match students looking for jobs w
...
 
 During each sprint, the tasks are implemented and merged with code reviews. There are regular meetings to share progress and resolve blockers. At the end of the sprint, the stories are reviewed to check if they are done or if further tasks need to be added. There is also a retrospective to give feedback and improve the process.
 
-The project is implemented in Java using Spring Boot, based on the Spring framework with a MySQL database. Maven is used as the build tool.
+The project is implemented in Java using the Spring Boot framework for the backend. MySQL is utilized as the database engine. Thymeleaf is employed for the interaction between the frontend and backend, while HTML, CSS, and JavaScript are used for frontend templates. Maven is the chosen build tool.
```
Comment (MajorTreble) [inline README.md:34]: MySQL is used ...

File: README.md [hunk 3]
```diff
@@ -27,9 +27,54 @@ The application is a career search web tool to match students looking for jobs w
...
 
+## Set up the Project
+
+Follow these steps to set up the project:
+
+1. Clone the project repository using the following command in the target local project folder: git clone https://github.com/MajorTreble/WhateverJobYouChoose-Soen6011summer2023.git
+
+2. Open the project in IntelliJ IDEA by selecting "Open" and navigating to the project directory.
```
Comment (MajorTreble) [inline README.md:42]: Not everyone uses IntelliJ
We should give the maven commands first.

File: README.md [hunk 4]
```diff
@@ -27,9 +27,54 @@ The application is a career search web tool to match students looking for jobs w
...
+
+3. In IntelliJ IDEA, open the `pom.xml` file. IntelliJ should recognize it as a Maven project and prompt you to import it. Click on "Open as Project" to import the project and download all the dependencies specified in the `pom.xml` file.
+
+4. Create a local MySQL server database using the credentials mentioned in the `application.yml` file. Make sure you have MySQL server installed. You can use MySQL Workbench or any other MySQL client tool to create the database.
```
Comment (MajorTreble) [inline README.md:46]: Be more specific.

File: README.md [hunk 5]
```diff
@@ -27,9 +27,54 @@ The application is a career search web tool to match students looking for jobs w
...
+
+5. Once the database is set up, create a schema named `soen6011` in the local server. This schema will be used by the application to store its data.
+
+6. Run the application in IntelliJ IDEA by clicking the green triangle button (Run). This will start the application and automatically create all the necessary database tables in the `soen6011` schema.
```
Comment (MajorTreble) [inline README.md:50]: Prefer maven commands

File: README.md [hunk 6]
```diff
@@ -27,9 +27,54 @@ The application is a career search web tool to match students looking for jobs w
...
+
+7. To add sample data to the database for testing the application, run the SQL query named `sample_data_to_push_into_database.sql`. This query contains predefined data that can be inserted into the tables. Execute the query in your MySQL client tool.
+
+Now you have successfully set up the project, created the database schema, and added sample data to the database. You can test the application and explore its features.
+
+## Key Features
+
+The project includes the following core features:
+
+1. **Employer**
```
Comment (MajorTreble) [inline README.md:60]: This could be written as a paragraph rather than a bullet point list.

**Comment by MajorTreble** [comment]
No unit tests
No javadoc

**Pull request merged**